src/java/org/apache/cassandra/dht/RangeStreamer.java

```diff
@@ -329,9 +330,13 @@ public void addRanges(String keyspaceName, ReplicaCollection<?> replicas)
             logger.info("{}: range {} exists on {} for keyspace {}", description, entry.getKey(), entry.getValue(), keyspaceName);
 
         Multimap<InetAddressAndPort, FetchReplica> workMap;
-        //Only use the optimized strategy if we don't care about strict sources, have a replication factor > 1, and no
-        //transient replicas.
-        if (useStrictSource || strat == null || strat.getReplicationFactor().allReplicas == 1 || strat.getReplicationFactor().hasTransientReplicas())
+        //Only use the optimized strategy if we don't care about strict sources, have a replication factor > 1, no
+        //transient replicas or HCD-84
+        if (Boolean.parseBoolean(System.getProperty("com.datastax.RangeStreamer.skipFFOptimization", "true")) ||
```

Comment:
:comment
I think it should be extracted to https://github.com/datastax/cassandra/blob/main/src/java/org/apache/cassandra/config/CassandraRelevantProperties.java, at least that's a pattern I have seen people using in CC.

:question
Also I wonder about the name of the property itself, it's `com.datastax.RangeStreamer.skipFFOptimization`. I wonder whether it could be something less implementation specific (after all FF can be changed to a different algorithm), maybe `skip_optimal_streaming_candidates_calculation` would be a bit more intuitive? There is also a matter of the prefix, shall we use `com.datastax` or go with `hcd`/`cassandra`?

Reply:
Icwym. I was seeing more like a ninja personalized property. But it can be promoted to a first class C* variable indeed.